nbs_probabilistic/create_dropout_ensemble.py

```diff
@@ -0,0 +1,131 @@
+    batch_size=args['batch_size']
+    output_vars=args['output_vars']
+    
+    #optional inputs. see load_args() for default values.
+    data_subsample=args['data_subsample']
+    norm_subsample=args['norm_subsample']
+    nt_in=args['nt_in'] #Ques: sometimes mentioned in config file as 'nt'. is that same?
+    dt_in=args['dt_in']
+    test_years=args['test_years']
+    lead_time=args['lead_time']
+    
+    #changing paths
+#   model_save_dir='/home/garg/data/WeatherBench/predictions/saved_models'
+#   datadir='/home/garg/data/WeatherBench/5.625deg'
+    
+    #run data generator
+    ds = xr.merge([xr.open_mfdataset(f'{datadir}/{var}/*.nc', combine='by_coords') 
+                   for var in var_dict.keys()])
+    mean = xr.open_dataarray(f'{model_save_dir}/{exp_id}_mean.nc') 
+    std = xr.open_dataarray(f'{model_save_dir}/{exp_id}_std.nc')
+    
+    ds_test= ds.sel(time=slice(test_years[0],test_years[-1]))
+    #Ques: Should shuffle be False? since its testing.
+    dg_test = DataGenerator(ds_test, var_dict, lead_time, batch_size=batch_size, shuffle=False,
+                            load=True,mean=mean, std=std, output_vars=output_vars,
+                            data_subsample=data_subsample, norm_subsample=norm_subsample,
+                            nt_in=nt_in,dt_in=dt_in) 
+    return dg_test
+
+def get_model(exp_id, model_save_dir):
+    tf.compat.v1.disable_eager_execution() #needed
+    saved_model_path=f'{model_save_dir}/{exp_id}.h5'
+    
+    substr=['resnet','unet_google','unet']
+    assert any(x in exp_id for x in substr)
+    model=tf.keras.models.load_model(saved_model_path,
+                                     custom_objects={'PeriodicConv2D':PeriodicConv2D,
+                                    'lat_mse': tf.keras.losses.mse})
+    #ToDo: add other loss functions to custom_objects. doesn't matter if it is not used in the model itself, only so that load_model() doesn't break)
+    #Since we dont build again, we dont need to pass model params like kernel, filters, activation, dropout,loss and other details to the network.
+
+    #model.summary()
+    return model
+
+def predict(dg_test,model,number_of_forecasts, output_vars):
+    #For just 1 batch of data
+#     X,y=dg_test[0] #currently limiting output due to RAM issues.
+    
+#     #test-time dropout
+#     func = K.function(model.inputs + [K.learning_phase()], model.outputs)
+#     pred_ensemble = np.array([np.asarray(func([X] + [1.]), dtype=np.float32).squeeze() for _ in
+#                               range(number_of_forecasts)])
+    
+    #For Full Data
+    #NOTE: Always better to append in a list, rather than as numpy array. bcoz numpy array equires contiguous memory, so if you keep appending it would be slow. alternatively, pre-allocate an empty array. Current method: append to list--> convert to numpy array-->reshape.
+    func = K.function(model.inputs + [K.learning_phase()], model.outputs)
+    preds = []
+    counter=0
+    for X, y in dg_test:
```

Comment:
Little coding tip to avoid the counter.

Option 1: 
```
for i, (X, y) in enumerate(dg_test):
   print(i)
```

Option 2 (my favorite!)
```
from tqdm import tqdm
for X, y in tqdm(dg_test):
   ...
```
This will automatically create a progress bar.
```